PhysicsTools/PatAlgos/plugins/BaseMVAValueMapProducer.h

```diff
@@ -0,0 +1,159 @@
+#ifndef PhysicsTools_PatAlgos_BaseMVAValueMapProducer
+#define PhysicsTools_PatAlgos_BaseMVAValueMapProducer
+
+// -*- C++ -*-
+//
+// Package:    PhysicsTools/PatAlgos
+// Class:      BaseMVAValueMapProducer
+// 
+/**\class BaseMVAValueMapProducer BaseMVAValueMapProducer.cc PhysicsTools/PatAlgos/plugins/BaseMVAValueMapProducer.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Andre Rizzi
+//         Created:  Mon, 07 Sep 2017 09:18:03 GMT
+//
+//
+
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+
+#include "TMVA/Factory.h"
+#include "TMVA/Reader.h"
+
+#include "CommonTools/Utils/interface/StringObjectFunction.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "CommonTools/Utils/interface/TMVAZipReader.h"
+
+#include <string>
+//
+// class declaration
+//
+
+template <typename T>
+class BaseMVAValueMapProducer : public edm::stream::EDProducer<> {
+   public:
+  explicit BaseMVAValueMapProducer(const edm::ParameterSet &iConfig):
+    src_(consumes<edm::View<T>>(iConfig.getParameter<edm::InputTag>("src"))),
+    variablesOrder_(iConfig.getParameter<std::vector<std::string>>("variablesOrder")),
+    name_(iConfig.getParameter<std::string>("name")),
+    isClassifier_(iConfig.getParameter<bool>("isClassifier"))  
+  {
+      edm::ParameterSet const & varsPSet = iConfig.getParameter<edm::ParameterSet>("variables");
+      for (const std::string & vname : varsPSet.getParameterNamesForType<std::string>()) {
+	  funcs_.emplace_back(std::pair<std::string,StringObjectFunction<T,true>>(vname,varsPSet.getParameter<std::string>(vname)));
+      }
+
+      values_.resize(variablesOrder_.size());
+      size_t i=0;
+      for(const auto & v : variablesOrder_){
+	positions_[v]=i;
+	reader_.AddVariable(v,(&values_.front())+i);
+	i++;
+      }
+//      reader_.BookMVA(name_,iConfig.getParameter<edm::FileInPath>("weightFile").fullPath() );
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant